qiskit/pulse/compiler/target.py

```diff
@@ -151,156 +163,94 @@ def __init__(
-
-        self._coupler_ports = coupler_ports
-        self._coupler_ports_inv = dict(zip(coupler_ports.values(), coupler_ports.keys()))
-
-        self._mixed_frames = mixed_frames
-
-    def is_mixed_frame_available(
-        self,
-        mixed_frame: model.MixedFrame,
-    ) -> bool:
-        if not isinstance(mixed_frame, model.MixedFrame):
-            raise TypeError(f"{mixed_frame} is not a MixedFrame object.")
-        try:
-            p_uid = self.get_port_identifier(mixed_frame.pulse_target)
-            f_uid = self.get_frame_identifier(mixed_frame.frame)
-        except NotExistingComponent:
-            return False
-        try:
-            return f_uid in self._mixed_frames[p_uid]
-        except KeyError:
-            return False
+        # Use table-like data format, if we can add to dependency.
+        # In realistic provider implementation use of schema is recommended.
+        self._tx_ports = tx_ports
 
     def get_frame_identifier(
         self,
         frame: model.Frame,
     ) -> str:
+        if isinstance(frame, model.GenericFrame):
+            return frame.name
         if isinstance(frame, model.QubitFrame):
             try:
                 return self._qubit_frames[frame.index]
             except KeyError as ex:
                 raise NotExistingComponent(
-                    "This hardware doesn't provide any frame for "
-                    f"QubitFrame of index {frame.index}."
+                    "This control system doesn't provide any frame for "
+                    f"implementing QubitFrame of index {frame.index}."
                 ) from ex
         if isinstance(frame, model.MeasurementFrame):
             try:
                 return self._meas_frames[frame.index]
             except KeyError as ex:
                 raise NotExistingComponent(
-                    "This hardware doesn't provide any frame for "
-                    f"MeasurementFrame of index {frame.index}."
+                    "This control system doesn't provide any frame for "
+                    f"implementing MeasurementFrame of index {frame.index}."
                 ) from ex
         raise TypeError(
-            f"Input frame type {frame.__class__.__name__} cannot "
-            "be directly mapped to hardware elements."
+            f"{self.__class__.__name__} doesn't recognize the frame object of type "
+            f"{frame.__class__.__name__}. If you are using a custom subclass, "
+            "you must also define PulseTarget class supporting this type."
         )
 
-    def get_port_identifier(
+    def get_generic_port_identifier(
         self,
-        port: model.LogicalElement,
+        signal_entry: model.SignalEntry,        
     ) -> str:
-        if isinstance(port, model.Qubit):
-            try:
-                return self._qubit_ports[port.qubit_index]
-            except KeyError as ex:
+        return self._get_port_common(signal_entry, "generic")
```

Comment:
I wonder if `_get_port_common` should be a public interface, maybe with a default value for `op_type`? It seems awkward to use a special method for measure ports and another for everything else. Also, it feels like a language mismatch to label the ports "generic" and "measure" while the frames are "Qubit", "Measure" and "Generic" -- for one qubits are "generic" but for the other they are distinct from "Generic". Maybe the qubit port op types should be "control" and "measure" or something like that?

Reply:
This is reasonable suggestion. Two methods are merged into a single method `get_port_identifier` which is aligned with frame query `get_frame_identifier`.